packages/common/src/services/extension.service.ts

```diff
@@ -62,6 +65,7 @@ export class ExtensionService {
     protected readonly sortService: SortService,
     protected readonly treeDataService: TreeDataService,
     protected readonly translaterService?: TranslaterService,
+    protected readonly lazyGridService?: () => GridService
```

Comment:
nice name since it really is lazy loaded, I'm still impressed by this approach 😉

Reply:
yeah lazy resolvers are a great thing over at C# standard DI and also great in Aurelia. Its a cool way to push initialization of transient tokens all the way to the point when they are actually needed.

In Angular its a bit cumbersome with Lazy Module loading but serves the same purpose.

the async nature on the other hand makes it a nice escape hatch for circular refs ;)